includes/class-newspack-popups-exporter.php

```diff
@@ -0,0 +1,263 @@
+	 * @return array The prompt in the format expected by the exporter.
+	 */
+	private function prepare_prompt_for_export( $prompt ) {
+		unset( $prompt['id'] );
+
+		if ( isset( $prompt['options']['excluded_categories'] ) ) {
+			$prompt['options']['excluded_categories'] = $this->sanitize_categories( $prompt['options']['excluded_categories'] );
+		}
+		if ( isset( $prompt['options']['excluded_tags'] ) ) {
+			$prompt['options']['excluded_tags'] = $this->sanitize_tags( $prompt['options']['excluded_tags'] );
+		}
+		if ( isset( $prompt['categories'] ) ) {
+			$prompt['categories'] = $this->sanitize_categories( $prompt['categories'] );
+		}
+		if ( isset( $prompt['tags'] ) ) {
+			$prompt['tags'] = $this->sanitize_tags( $prompt['tags'] );
+		}
+		if ( isset( $prompt['campaign_groups'] ) ) {
+			$prompt['campaign_groups'] = $this->sanitize_campaign_groups( $prompt['campaign_groups'] );
+		}
+
+		// There was a bug that was saving some overlay_sizes as full instead of full-width. Let's take this into account and fix it.
+		if ( isset( $prompt['options']['overlay_size'] ) && 'full' === $prompt['options']['overlay_size'] ) {
+			$prompt['options']['overlay_size'] = 'full-width';
+		}
+
+		if ( isset( $prompt['options']['utm_suppression'] ) && empty( $prompt['options']['utm_suppression'] ) ) {
+			unset( $prompt['options']['utm_suppression'] );
+		}
+
+		return $prompt;
+
+	}
+
+	/**
+	 * Sanitizes an array of Categories
+	 *
+	 * @see self::sanitize_terms
+	 *
+	 * @param int[]|WP_Term[] $categories An array of categories IDs or WP_Term objects.
+	 * @return array
+	 */
+	private function sanitize_categories( $categories ) {
+		return $this->sanitize_terms( $categories, 'category' );
+	}
+
+	/**
+	 * Sanitizes an array of Tags
+	 *
+	 * @see self::sanitize_terms
+	 *
+	 * @param int[]|WP_Term[] $tags An array of tags IDs or WP_Term objects.
+	 * @return array
+	 */
+	private function sanitize_tags( $tags ) {
+		return $this->sanitize_terms( $tags, 'post_tag' );
+	}
+
+	/**
+	 * Sanitizes an array of Campaigns
+	 *
+	 * @see self::sanitize_terms
+	 *
+	 * @param int[]|WP_Term[] $campaign_groups An array of campaign_groups IDs or WP_Term objects.
+	 * @return array
+	 */
+	private function sanitize_campaign_groups( $campaign_groups ) {
+		return $this->sanitize_terms( $campaign_groups, Newspack_Popups::NEWSPACK_POPUPS_TAXONOMY );
+	}
+
+	/**
+	 * Sanitize terms.
+	 *
+	 * Transforms an array of terms in an array in the format expected by the Exporter, containing only the term ID and name.
+	 *
+	 * @param int[]|WP_Term[] $terms Array of terms. If term must be either an integer with the term ID, or a WP_Term object.
+	 * @param string          $taxonomy Taxonomy slug. Used to fetch the term name if only an ID is informed.
+	 * @return array
+	 */
+	private function sanitize_terms( $terms, $taxonomy ) {
+		$sanitized_terms = [];
+		foreach ( $terms as $term ) {
```

Comment:
For this line, the following warning is logged: `PHP Warning:  foreach() argument must be of type array|object, bool given`